### ISSUE TITLE
Functionality in Labeling and Predicting

### ISSUE THREAD

**Issue by dbedi3311** [jbohnslav/deepethogram#23]
I have a really long video that I intend to train using deepethogram. My ideal work-flow would consist of me only labeling _some_ frames in the entire video with their respective behaviors rather than needing to label _every_ single frame in the video. Using this, I would want deepethogram to predict the unlabeled frames (i.e. the ones that I didn't label already). However, deepethogram automatically sets frames as background if they are unlabeled during the process of finalizing frames. Does the model train on the these background behaviors; in effect, does it learn to classify behaviors as label 0: background, or does it revert a frame to background if the thresholds aren't met for any of the non-background behaviors? It's a big hindrance needing to have a fully annotated video, so what do you suggest I do?

**Comment by jbohnslav** [comment]
Hi @dbedi3311, this is an interesting request. 

> My ideal work-flow would consist of me only labeling some frames in the entire video with their respective behaviors rather than needing to label every single frame in the video. 

There are two reasons I didn't allow this workflow. One: keeping our validation and test data held-out. Two: sequence models. 

We keep track of train, validation, and test splits by video, rather than by frame. This makes the entire pipeline much simpler, as we do not have to keep track of every frame for all of our videos for our splits of the data. This also enforces that our validation and test sets are entirely different videos-- in my case, this means that our held out data is from different animals as well. 
 
For sequence models, our inputs and targets are sequences, say `sequence_length = 180` timepoints. We could either first comb through the dataset to look for contiguously labeled chunks of `sequence_length` frames, or we could mask the sequence loss for unlabeled data. However, (going back to point one) if we have interspersed train and validation labels, we can no longer easily sample from our sequences. 

We could enforce train / validation splits in chunks rather than truly interspersed--however, the chunk size will be related to the `sequence_length` hyperparameter which is at a much later step than labeling the data. 

If this is a very popular feature request, I can look into fixing these problems and reopen this request. Until then, I recommend you split your very long video into chunks and fully label each chunk. Feel free to comment / ask questions even though I'm closing the issue.
